test/Microsoft.Framework.Logging.Test/LoggerExtensionsTest.cs

```diff
@@ -591,6 +592,28 @@ public void LogValuesEventIdAndError_LogsCorrectValues()
                 debug.Formatter(debug.State, debug.Exception));
         }
 
+        [Fact]
+        public void BeginScope_CreatesScope_WithFormatStringValues()
+        {
+            // Arrange
+            var testSink = new TestSink(
+                writeEnabled: (writeContext) => true, 
+                beginEnabled: (beginScopeContext) => true);
+            var logger = new TestLogger("TestLogger", testSink, enabled: true);
+            var actionName = "App.Controllers.Home.Index";
+            var expectedStringMessage = "Executing action " + actionName;
+
+            // Act
+            var scope = logger.BeginScope("Executing action {ActionName}", actionName);
+
+            // Assert
+            Assert.Equal(1, testSink.Scopes.Count);
+            var scopeState = testSink.Scopes[0].Scope as FormattedLogValues;
+            Assert.NotNull(scopeState);
```

Comment:
Assert.IsType does the cast for you, and gives you a good error message, Assert.Null does not. Use the right tool for the job.